### PR TITLE
feat(cactus-plugin-ledger-connector-ethereum): refactor connector API

### PR DESCRIPTION
**Pull request by outSH**
- Refactor to single `invokeContract` endpoint that can accept multiple methods of 
    supplying contract definition (directly, from keychain, etc..).
- Same for `deployContract` method.
- Update cactus-common `safeStringifyException` function to better handle axios errors
    (and other custom exceptions that support `toJSON()` method.).
- Use common error handling in all connector endpoints (except prometheus).
- Fix the tests.
- Switch to `Cacti` work in entire connector (except for package name)

Depends on #2630

Signed-off-by: Michal Bajer <michal.bajer@fujitsu.com>

### CODE REVIEW

File: packages/cactus-common/src/main/typescript/error-utils.ts [hunk 1]
```diff
@@ -28,5 +28,16 @@ export function safeStringifyException(error: unknown): string {
...
+  // Use it if available
+  if (
+    error &&
+    typeof error === "object" &&
+    "toJSON" in error &&
+    typeof error.toJSON === "function"
+  ) {
+    error = error.toJSON();
+  }
+
```
Comment (petermetz) [inline packages/cactus-common/src/main/typescript/error-utils.ts:41]: @outSH I agree with the idea but I have one nit-pick:
If the execution hits this if-condition then `error` will be a JSON string and then we call `safeStringify()` on it which will mangle that by escaping everything in it (all the double quotes, etc.).

Not really a big issue but it's definitely one of those quality of life things that we could improve for people reading the crashlogs.

TLDR: I would refactor it in a way that safeStringify is only called on the else (e.g. if there was no `toJSON()` implementation found on the error)

File: packages/cactus-plugin-ledger-connector-ethereum/src/main/json/openapi.json [hunk 2]
```diff
@@ -0,0 +1,1164 @@
...
+  "openapi": "3.0.3",
+  "info": {
+    "title": "Hyperledger Cacti Plugin - Connector Ethereum",
+    "description": "Can perform basic tasks on a Ethereum ledger",
+    "version": "0.0.1",
+    "license": {
+      "name": "Apache 2.0",
+      "url": "https://www.apache.org/licenses/LICENSE-2.0.html"
+    }
+  },
```
Comment (petermetz) [inline packages/cactus-plugin-ledger-connector-ethereum/src/main/json/openapi.json:11]: @outSH There are some new custom-checks in the CI that might start failing because of the version not matching the current release. Just a note for future reference so you know if it happens!

Comment (outSH) [reply packages/cactus-plugin-ledger-connector-ethereum/src/main/json/openapi.json:11]: Copied this comment to dependent PR to fix it there - https://github.com/hyperledger/cacti/pull/2581

File: packages/cactus-common/src/main/typescript/error-utils.ts (hunk 1 above)
Comment (outSH) [reply packages/cactus-common/src/main/typescript/error-utils.ts:41]: @petermetz Good catch - fixed

File: packages/cactus-plugin-ledger-connector-ethereum/src/main/typescript/plugin-ledger-connector-ethereum.ts [hunk 3]
```diff
@@ -0,0 +1,998 @@
...
+        `Invalid method name provided in request. ${req.methodName} does not exist on the Web3 contract object's "methods" property.`,
+      );
+    }
+
+    const methodRef = contractInstance.methods[req.methodName] as (
+      ...args: unknown[]
+    ) => PayableMethodObject;
+    Checks.truthy(methodRef, `${fnTag} YourContract.${req.methodName}`);
+
+    const method = methodRef(...req.params);
```
[the remark this answers is not in the document]
Comment (outSH) [reply packages/cactus-plugin-ledger-connector-ethereum/src/main/typescript/plugin-ledger-connector-ethereum.ts]: Leftover from dependency commit - false positive

File: packages/cactus-plugin-ledger-connector-ethereum/src/main/typescript/plugin-ledger-connector-ethereum.ts [hunk 4]
```diff
@@ -0,0 +1,998 @@
...
+    if (!isSafeToCall) {
+      throw new RuntimeError(
+        `Invalid method name provided in request. ${args.contractMethod} does not exist on the Web3 contract object's "methods" property.`,
+      );
+    }
+
+    const methodRef = contract.methods[args.contractMethod] as (
+      ...args: unknown[]
+    ) => any;
+    return methodRef(...contractMethodArgs)[args.invocationType](
```
[the remark this answers is not in the document]
Comment (outSH) [reply packages/cactus-plugin-ledger-connector-ethereum/src/main/typescript/plugin-ledger-connector-ethereum.ts]: Leftover from dependency commit - false positive

File: packages/cactus-plugin-ledger-connector-ethereum/src/main/typescript/plugin-ledger-connector-ethereum.ts [hunk 5]
```diff
@@ -0,0 +1,998 @@
...
+        args.methodName,
+      );
+    if (!isSafeToCall) {
+      throw new RuntimeError(
+        `Invalid method name provided in request. ${args.methodName} does not exist on the Web3.Eth object.`,
+      );
+    }
+
+    const web3MethodArgs = args.params || [];
+    return looseWeb3Eth[args.methodName](...web3MethodArgs);
```
[the remark this answers is not in the document]
Comment (outSH) [reply packages/cactus-plugin-ledger-connector-ethereum/src/main/typescript/plugin-ledger-connector-ethereum.ts:1018]: Leftover from dependency commit - false positive

**Review by petermetz — APPROVED** [review]
LGTM

**Review by izuru0 — APPROVED** [review]
LGTM
